src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -459,9 +460,21 @@ void GcsNodeManager::StartNodeFailureDetector() {
 
 void GcsNodeManager::UpdateNodeRealtimeResources(
     const ClientID &node_id, const rpc::HeartbeatTableData &heartbeat) {
-  auto resources_available = MapFromProtobuf(heartbeat.resources_available());
-  cluster_realtime_resources_[node_id] =
-      std::make_shared<ResourceSet>(resources_available);
+  if (light_heartbeat_enabled_) {
+    if (heartbeat.resources_available_size() > 0) {
+      if (heartbeat.resources_available().count(kResourcesTurningEmpty) == 1) {
+        cluster_realtime_resources_[node_id] = std::make_shared<ResourceSet>();
+      } else {
+        auto resources_available = MapFromProtobuf(heartbeat.resources_available());
```

Comment:
Based on your description, isn't `resources_available ` just an empty resourceSet?

Reply:
Not always. If resource_available_size is not empty, it has two scenarios: 1. only contains "TURNING_EMPTY", which means available resources in this node become empty from non-empty. 2. contains number of resource key/value like { "cpu": 1, "gpu":2, "memory": 1024 }, this means available resources has changed in this node and it report newest available resources in heartbeat.